llvm/lib/CodeGen/GlobalISel/LegalizerHelper.cpp

```diff
@@ -6173,8 +6173,7 @@ LegalizerHelper::equalizeVectorShuffleLengths(MachineInstr &MI) {
     // Extend mask to match new destination vector size with
     // undef values.
     SmallVector<int, 16> NewMask(Mask);
-    for (unsigned I = MaskNumElts; I < SrcNumElts; ++I)
-      NewMask.push_back(-1);
+    NewMask.resize(SrcNumElts, -1);
```

Comment:
Use the constructor variant as you did in `moreElementsVectorShuffle()`?

Reply:
We're already using the constructor to copy the first part of the Mask. We could do

```
SmallVector<int, 16> NewMask(SrcNumElts, -1);
std::copy(NewMask.begin(), Mask.begin(), Mask.end())
```

Then we'd have the correct size at the start.